### PR TITLE
Add QualityForRecognition to Face Detection

### PR DESCRIPTION
**Pull request by msyache**
Added the optional attribute QualityForRecognition to Face Detection API.
Also includes some document discrepancy bugfixes.
Related swagger change PR: 
https://github.com/Azure/azure-rest-api-specs/pull/14556
https://github.com/Azure/azure-rest-api-specs/pull/15187
https://github.com/Azure/azure-rest-api-specs/pull/16449
https://github.com/Azure/azure-rest-api-specs/pull/16846

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Review by lulululululu — APPROVED** [review]

File: sdk/cognitiveservices/Vision.Face/src/Generated/Models/PersonGroup.cs [hunk 1]
```diff
@@ -30,16 +30,16 @@ public PersonGroup()
...
         /// <param name="name">User defined name, maximum length is
         /// 128.</param>
+        /// <param name="personGroupId">PersonGroupId of the target person
+        /// group.</param>
         /// <param name="userData">User specified data. Length should not
         /// exceed 16KB.</param>
         /// <param name="recognitionModel">Possible values include:
         /// 'recognition_01', 'recognition_02', 'recognition_03',
         /// 'recognition_04'</param>
-        public PersonGroup(string personGroupId, string name = default(string), string userData = default(string), string recognitionModel = default(string))
```
Comment (dipidoo) [inline sdk/cognitiveservices/Vision.Face/src/Generated/Models/PersonGroup.cs:42]: should the order be preserved? this feels inconveniently silent for people who upgrades

File: sdk/cognitiveservices/Vision.Face/src/Generated/Models/LargePersonGroup.cs [hunk 2]
```diff
@@ -30,16 +30,16 @@ public LargePersonGroup()
...
         /// 128.</param>
+        /// <param name="largePersonGroupId">LargePersonGroupId of the target
+        /// large person groups</param>
         /// <param name="userData">User specified data. Length should not
         /// exceed 16KB.</param>
         /// <param name="recognitionModel">Possible values include:
         /// 'recognition_01', 'recognition_02', 'recognition_03',
         /// 'recognition_04'</param>
-        public LargePersonGroup(string largePersonGroupId, string name = default(string), string userData = default(string), string recognitionModel = default(string))
+        public LargePersonGroup(string name, string largePersonGroupId, string userData = default(string), string recognitionModel = default(string))
```
Comment (dipidoo) [inline sdk/cognitiveservices/Vision.Face/src/Generated/Models/LargePersonGroup.cs:42]: same comment here as below

File: sdk/cognitiveservices/Vision.Face/src/Generated/Models/LargeFaceList.cs [hunk 3]
```diff
@@ -30,16 +30,16 @@ public LargeFaceList()
...
         /// 128.</param>
+        /// <param name="largeFaceListId">LargeFaceListId of the target large
+        /// face list.</param>
         /// <param name="userData">User specified data. Length should not
         /// exceed 16KB.</param>
         /// <param name="recognitionModel">Possible values include:
         /// 'recognition_01', 'recognition_02', 'recognition_03',
         /// 'recognition_04'</param>
-        public LargeFaceList(string largeFaceListId, string name = default(string), string userData = default(string), string recognitionModel = default(string))
+        public LargeFaceList(string name, string largeFaceListId, string userData = default(string), string recognitionModel = default(string))
```
Comment (dipidoo) [inline sdk/cognitiveservices/Vision.Face/src/Generated/Models/LargeFaceList.cs:42]: same comment here as below

File: sdk/cognitiveservices/Vision.Face/src/Generated/Models/FaceList.cs [hunk 4]
```diff
@@ -32,18 +32,18 @@ public FaceList()
...
+        /// list.</param>
         /// <param name="userData">User specified data. Length should not
         /// exceed 16KB.</param>
         /// <param name="recognitionModel">Possible values include:
         /// 'recognition_01', 'recognition_02', 'recognition_03',
         /// 'recognition_04'</param>
         /// <param name="persistedFaces">Persisted faces within the face
         /// list.</param>
-        public FaceList(string faceListId, string name = default(string), string userData = default(string), string recognitionModel = default(string), IList<PersistedFace> persistedFaces = default(IList<PersistedFace>))
+        public FaceList(string name, string faceListId, string userData = default(string), string recognitionModel = default(string), IList<PersistedFace> persistedFaces = default(IList<PersistedFace>))
```
Comment (dipidoo) [inline sdk/cognitiveservices/Vision.Face/src/Generated/Models/FaceList.cs:46]: same comment here as below

**Review by dipidoo — DISMISSED** [review]
There are some reordered parameters of same type (string) that may be confusing for people who upgrades

**Review by dipidoo — APPROVED** [review]

**Review by jsquire — APPROVED** [review]

**Pull request merged**